### PR TITLE
Javascript Library

### PR DESCRIPTION
**Pull request by Jhonatanssal**
This is a project of the Javascript module. It is preceded by the lesson about Objects and the Object Constructor.
The library work with a form that gets the data and create the DOM elements in the file.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,38 +1,83 @@
...
+function addBook(e) {
+  e.preventDefault();
+  const title = document.getElementById('title').value;
+  const author = document.getElementById('author').value;
+  const pages = document.getElementById('pages').value;
+  const status = document.getElementById('read').checked;
+
+  const book = new Book(title, author, pages, status);
+
+  create(book, booksContainer);
```
Comment (MuriloRoque) [inline app.js]: - [x] Your function `addBookToLibrary` only pushes a book into an array, why don't you just delete that function and put the statement inside the `addBook` function. Also, rename the function `addBook` to `addBookToLibrary`, so we can be on par with the requirements ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/book_array_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/library))

File: index.html [hunk 2]
```diff
@@ -3,21 +3,27 @@
...
+            <div class="form-group bg-light col-3 mx-auto p-3">
+                <label for='title'>Book name</label><br>
+                <input type='text' id='title' placeholder="Book title" required><br><br>
+                <label for='author'>Author name</label><br>
+                <input type='text' id='author' placeholder='Author name' required><br><br>
+                <label for='pages'>Number pages</label><br>
+                <input type='number' id='pages' placeholder='Number of pages' required><br><br>
+                <label for='read'>Read</label>
+                <input type='checkbox' id='read'><br>
+                <input type='submit' id='button'>
```
Comment (MuriloRoque) [inline index.html]: - [x] Change the name of the button from `Enviar` to `New Book`, so everything is in English 😉 (P.S.: It looks like we have a Brazilian here, "Olá, meu amigo! 😄")

File: app.js [hunk 3]
```diff
@@ -1,38 +1,83 @@
...
+Book.prototype.changeStatus = function changeStatus() {
+  if (this.status === true) {
+    this.status = false;
+  } else {
+    this.status = true;
+  }
+};
 
-let myLibrary = [];
+function create(book) {
```
Comment (MuriloRoque) [inline app.js]: - [x] This function should loop through the `myLibrary` array and display each book in your book container, you are just creating the elements in your page. For example, why did you create the array if you are not using it for anything but push books inside? ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/books_render_loop.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/library))

- [x] We also need a function to delete a book from the array ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/remove_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/library))

File: app.js [hunk 4]
```diff
@@ -1,38 +1,83 @@
...
+  cont.appendChild(readBtn);
 
-    
+  readBtn.addEventListener('click', () => {
+    book.changeStatus();
+    if (book.status === true) {
+      readBtn.textContent = 'Not Read';
+    } else if (book.status === false) {
+      readBtn.textContent = 'Read';
+    }
```
Comment (kubilaycaglayan) [inline app.js]: :bulb: This is a little part of application logic. And it is happening in the DOM manipulation method. It is better to make an isolated function to change the status of a specific book in the array and call it here.

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
#### Hello Team 👋
#### Status: Changes Required ♻️

Great job so far! 👏👋
Let's cover a few details together to make your project better 😊⏬

## GitFlow

- [x] [REQUIRED] Set your `development` branch as default.

## DOM manipulation vs. Application Logic

- [x] [REQUIRED] Keep the application logic separated from DOM manipulation tasks ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png) from Pathwright)

💡 Check the inline comments below please.

## Display Book

Okay, this is mostly about the usage of the `booksContainer` array. I will give you some additional information, follow the bulbs 💡 please :blush:.

- [x] [REQUIRED] It has a function that loops through the array and displays each book ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/books_render_loop.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/library))

💡 Currently, you have an array `booksContainer` to store the books but you are not using that array to show the books.
You are creating components at the moment the submit button clicked. This fine but the components(containers) or book cards should be created by looping through the `booksContainer` array.

- [x] [REQUIRED] It displays on the page every book contained in the array ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/render_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/library))

💡 In other words, if we initialize the application with this array `const myLibrary = [new Book('title', 'author', 'pages', 'status')];` we should be able to see that book at the beginning.

- [x] [REQUIRED] Each `Book` displayed has a `button` to `remove` the `Book` from the `Library` array ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/remove_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/library))

💡 The idea is similar here. We need a button to remove the books from the array, therefore from the view.

- [x] [REQUIRED] Each `Book` displayed has a `button` to `toggle` the `Book` read status ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/toggle_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/library))

💡 Currently, we have this button but it should alter the objects in the array.

#### :arrow_right:  Imagine that array(`myLibrary`) is your Database and whenever you need to make a change you have to first change the Database.

## Here are some optional requirements if you want to go further and learn more 😇

- [x] [OPTIONAL] the form used to create a book should have validations, you should not be able to create an empty book

- [x] [OPTIONAL] It has data persistence by saving, retrieving, and updating the library books using [the localstorage API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/localstorage_stretch_requirement.png) from [Odin Project](https://www.theodinproject.com/courses/javascript/lessons/library))

## That's it, you can submit another code review request after making these changes ♻️ 😇

With all my respect 🙏
> Kubilay Caglayan | Technical Support Engineer

**Review by MuriloRoque — CHANGES_REQUESTED** [review]
## Status: Changes requested 🔴 

### Hello again @jssarrazolaa and @Alexoid1 👋 

### You have started really well, your page is almost perfect, we just need to fix some details before moving forward, alright?

- [x] Please check my comments in your code.

- [x] Add a nice screenshot to your README file.

- [x] In general, your functions are doing their job, but we need some refactoring:

- Separate your functions into more small functions, so they are not too big and hard to read. The below requirement will probably help with that.
- Keep your application logic separated from the DOM manipulation tasks ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png) from Pathwright). This is a hard one, but I will try helping you here. Create another JS file for DOM manipulation functions, like `document.getElementById` or `document.createElement`. In your main file, keep only logic related code that won't 'mess up' with the DOM, call these functions from your DOM manipulation file.

## Summary

- Your repository is very professional, keep up the good work 💪 
- Your page is doing what it needs to do, except for deleting a book 🚀 
- We just need to work a little more on the requirements I mentioned and we will be ready to go 👍 

Kindly make the requested changes and submit for another review, please :pray: 

I will be on Slack 24/7 in case you have any questions about this review :handshake: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow

**Comment by MuriloRoque** [comment]
### Hello again @jssarrazolaa and @Alexoid1 👋 

### I know you got two reviews from two different TSEs, it was probably a bug in our Dashboard. 

### But don't worry, we will try merging our reviews so you get only one list of requirements, can you please wait a little bit?

### Thank you

**Comment by MuriloRoque** [comment]
### Hello @jssarrazolaa and @Alexoid1 👋 

### We are very sorry about this bug, but we have decided to leave both our feedbacks in your Pull Request 😄 

### Since Kubilay suggested changes first, you may only follow his suggestions (for fairness), BUT it would be a huge plus if you could follow mine too, so your project can get even better!

### Happy coding! 💻

**Review by bellom — APPROVED** [review]
## Hi Team :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Sorry for the bug as the previous reviewer mentioned! 🙏 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next project :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**